compliance_checker/ioos.py

```diff
@@ -358,6 +358,10 @@ class IOOS1_2_ConventionsValidator(base.RegexValidator):
     validator_regex = r"\bIOOS-1.2\b"
     validator_fail_msg = "{} must contain the string \"IOOS 1.2\""
 
+class IOOS1_2_PlatformIDValidator(base.RegexValidator):
+    validator_regex = r"^\w+$"
```

Comment:
This is almost right if taking the regex approach, but `\w` also includes underscores.

From Python docs:

>  \w
>
>    Matches any alphanumeric character; this is equivalent to the class [a-zA-Z0-9_].

So use something like `[a-zA-Z0-9]` in place of `\w`